src/Client/widgets/Task.fs

```diff
@@ -10,157 +10,226 @@ open Fable.FontAwesome
 
     let handleChangeAnswer (event: FormEvent) =
-        dispatch (SetInput !!event.target?value)
-           
-    let handleKeyDown (event: KeyboardEvent) =
-        match event.keyCode with
-        | Keyboard.Codes.enter when model.Word.IsSome ->
-            match event.shiftKey with
-            | false -> dispatch SubmitAnswer
-            | true  -> dispatch UpdateTask
-        | Keyboard.Codes.ctrl ->
-            match event.shiftKey with
-            | false -> ()
-            | true  -> dispatch ShowAnswer
-        | _ -> 
-            ()
+        dispatch (SetAnswer !!event.target?value)
+               
+    Columns.columns [ Columns.IsGap (Screen.All, Columns.Is3) ]
+        [
+            Column.column [ ] [Tag.tag [ Tag.Color IsLight ; Tag.CustomClass "task-label" ] [inputViewState.Word] ]
+            Column.column [ ] [
+                                div [ClassName ("control has-icons-right " + inputViewState.InputClass)]
+                                    [
+                                        Input.text
+                                            [
+                                                Input.Props [OnChange handleChangeAnswer; OnKeyDown handleKeyDown; AutoCapitalize "none"] 
+                                                Input.Value inputViewState.InputText
+                                                Input.Size Size.IsLarge
+                                            ]
+                                        Icon.icon [ Icon.Size IsSmall; Icon.IsRight ]
+                                            [ Fa.i [ inputIcon ] [] ]
+                                    ]                     
+                              ]
+        ]
+
+type ButtonViewState = {
+    NextBtnDisabled : bool
+    CheckBtnDisabled : bool
+    ShowBtnDisabled : bool
+}
 
+let buttonView model dispatch nextButtonDisplayed =
     let handleShowAnswerClick _ = dispatch ShowAnswer
-    let handleUpdateClick _ = dispatch UpdateTask
-    let handleCheckClick _ = dispatch SubmitAnswer
-    
-    let nextButtonDisabled = not model.Word.IsSome
-    let checkButtonDisabled = model.Word.IsNone || model.AnswerShown
+    let handleUpdateClick _ = dispatch NextQuestion
+    let handleCheckClick _ = dispatch CheckAnswer
+
+    let buttonViewState =
+        match model.State with
+        | Fetching -> 
+            { NextBtnDisabled = true; CheckBtnDisabled = true; ShowBtnDisabled = true; }
+        | InputProvided _ ->
+            { NextBtnDisabled = false; CheckBtnDisabled = false; ShowBtnDisabled = false; }
+        | _ -> 
+            { NextBtnDisabled = false; CheckBtnDisabled = true; ShowBtnDisabled = false; }
+
+    let taskButton color handler text disabled = 
+        button IsMedium color handler text  
+            [
+                Button.Disabled disabled
+                Button.CustomClass "task-button"
+            ]
+
+    let rightButton = 
+        match nextButtonDisplayed with
```

Comment:
Now we'll have Check/Next button interchange, so there's no longer need to press shift button (only in case you want to show the answer) and if you want just to skip it you have to go throught Show -> Next button clicking, but I think its a good trade-off for good UX

Reply:
In general, I agree that it does simplify and improve UX and also looks cool.
One point is that the "way" to the "next task" is not obvious from the first glance now... I would prefer users to somehow know about this option.

Reply:
I think it makes sense to show the answer instead of just skipping the word, so once the user sees how the flow goes it will be obvious for them.

Reply:
We can try doing some testing on users :)

Reply:
Alright, that's fair. Last thing I want to ask - is it easy to do that on mobile the web does not show the keyboard shortcuts in the button text (just `Show`, `Check`, Next`)?

Reply:
And well, another thing... Now when the user changes the filter, they kind of have to do this "show/next" thing which is not a little confusing as well :(
Is it reasonable to add events for loading a new word when filter changes?

Reply:
> Alright, that's fair. Last thing I want to ask - is it easy to do that on mobile the web does not show the keyboard shortcuts in the button text (just `Show`, `Check`, Next`)?

For this I don't see an easy approach, but yeah it's bothering me as well



> And well, another thing... Now when the user changes the filter, they kind of have to do this "show/next" thing which is not a little confusing as well :(
> Is it reasonable to add events for loading a new word when filter changes?
I think this was working like that before or I did break it inadvertently. Anyway I can try making it reload with filters applied

Reply:
> For this I don't see an easy approach, but yeah it's bothering me as well

Alright, I will create a task then.

> I think this was working like that before or I did break it inadvertently. Anyway I can try making it reload with filters applied

Yes, it was like this before, it is just that now, without an explicit "next" button, it becomes less convenient to change the filters...

Reply:
I have added task reload on filter change. Check, please, if it's what you expect

Reply:
Yes, totally!